pkg/polaris/polaris_test.go

```diff
@@ -983,3 +987,590 @@ func TestGcpPermissions(t *testing.T) {
+	}
+	testServiceAccount, err := DefaultServiceAccount(true)
+	// Load configuration and create client. Usually resolved using the
+	// environment variable RUBRIK_POLARIS_SERVICEACCOUNT_FILE.
+	//polAccount, err := DefaultServiceAccount(true)
+	//if err != nil {
+	//	t.Fatal(err)
+	//}
+
+	client, err := NewClient(ctx, testServiceAccount, &polaris_log.DiscardLogger{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	slaDomains, err := client.K8s().ListSLA(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	basicCheck := make(map[string]bool)
+	for _, slaDomain := range slaDomains {
+		if slaDomain.Name == "Gold" || slaDomain.Name == "Silver" || slaDomain.Name == "Bronze" {
+			basicCheck[slaDomain.Name] = true
+		}
+	}
+	if !basicCheck["Gold"] || !basicCheck["Silver"] || !basicCheck["Bronze"] {
+		t.Errorf("failed to list default SLAs: %v", basicCheck)
+	}
+}
+
+// TestListSLA verifies that the SDK can list the default SLAs - Gold, Silver
+// and Bronze.
+//
+// To run this test against a Polaris instance the following environment
+// variables needs to be set:
+//   * TEST_INTEGRATION=1
+//   * RUBRIK_POLARIS_SERVICEACCOUNT_FILE=<path-to-polaris-service-account-file>
+func TestListK8sNamespace(t *testing.T) {
+	//if !boolEnvSet("TEST_INTEGRATION") {
+	//	t.Skipf("skipping due to env TEST_INTEGRATION not set")
+	//}
+
+	testClusterID := uuid.MustParse("2a47d1f1-0236-4030-87d4-837d2a75b370")
+	ctx := context.Background()
+	testServiceAccount := &ServiceAccount{
+		ClientID:       "client|sIIw3uAxHqFsn3kUR78AUf1zMewyLB7p",
+		ClientSecret:   "WnmUX2luK5X_TcrMMzZUrFh-mU7gWWti0VS90onJ_uwygXsYajUwVOlWE1MArIs_",
+		Name:           "test",
+		AccessTokenURI: "https://manifest.dev-045.my.rubrik-lab.com/api/client_token",
+	}
+	testServiceAccount, err := DefaultServiceAccount(true)
+	// Load configuration and create client. Usually resolved using the
+	// environment variable RUBRIK_POLARIS_SERVICEACCOUNT_FILE.
+	//polAccount, err := DefaultServiceAccount(true)
+	//if err != nil {
+	//	t.Fatal(err)
+	//}
+
+	client, err := NewClient(ctx, testServiceAccount, &polaris_log.DiscardLogger{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	nss, err := client.K8s().ListK8sNamespace(ctx, testClusterID)
+	if err != nil {
+		t.Fatal(err)
+	}
+	for _, ns := range nss {
+		fmt.Printf("%v\n", ns)
+		// all returned values should match the filter
+		if ns.K8sClusterID != testClusterID {
+			t.Fatalf(
+				"failed to filter based on clusterID: found clusterID %s, but expected %s",
+				ns.K8sClusterID,
+				testClusterID,
+			)
+		}
+	}
+
+}
+
+func TestUnassignSLA(t *testing.T) {
+	//if !boolEnvSet("TEST_INTEGRATION") {
+	//	t.Skipf("skipping due to env TEST_INTEGRATION not set")
+	//}
+
+	testNSID := uuid.MustParse("3b3d22b7-385c-5865-bd8f-0ff7534db42b")
+	ctx := context.Background()
+	testServiceAccount := &ServiceAccount{
+		ClientID:       "client|YVaYtseZQXRSiwEHV8HaRKfiRfsU0BhX",
+		ClientSecret:   "BS0-2olXRhY51R1AW0qj9gdgYBOs6x4uJbUg-7DXIxQImigEiavo819R0ZTPwq8a",
```

Comment:
:exclamation:**Cycode**: Secret of type: 'Generic Password' was found.
Severity: High <!-- 57ebe673-2cc2-4f77-9ece-c2f3ced36449 -->